sync.go

```diff
@@ -255,36 +317,40 @@ func (lw *LibWallet) RescanBlocks() error {
 		progress := make(chan wallet.RescanProgress, 1)
 		ctx, _ := contextWithShutdownCancel(context.Background())
 
-		var totalHeight int32
+		var totalHeightRescanned int32
 		go lw.wallet.RescanProgressFromHeight(ctx, netBackend, 0, progress)
 
 		for p := range progress {
 			if p.Err != nil {
 				log.Error(p.Err)
-
 				return
 			}
-			totalHeight += p.ScannedThrough
+			totalHeightRescanned += p.ScannedThrough
 			for _, syncProgressListener := range lw.syncProgressListeners {
-				syncProgressListener.OnRescan(p.ScannedThrough, SyncStateProgress)
+				report := &HeadersRescanProgressReport{
+					CurrentRescanHeight: p.ScannedThrough,
+					TotalHeadersToScan:  lw.GetBestBlock(),
+				}
+				syncProgressListener.OnHeadersRescanProgress(report)
 			}
-		}
 
-		select {
-		case <-ctx.Done():
-			for _, syncProgressListener := range lw.syncProgressListeners {
-				syncProgressListener.OnRescan(totalHeight, SyncStateProgress)
-			}
-		default:
-			for _, syncProgressListener := range lw.syncProgressListeners {
-				syncProgressListener.OnRescan(totalHeight, SyncStateFinish)
+			select {
+			case <-ctx.Done():
+				log.Info("Rescan cancelled through context")
+				return
```

Comment:
Errors such as these (in this case, the error is `ctx.Err()`) and the `p.Err` at the start of each loop cycle are currently ignored. Should find a way to notify the app of such errors so the interface(s) are updated properly. I've created an issue for this, to be worked on later: https://github.com/raedahgroup/dcrlibwallet/issues/35

We'd probably be better off introducing a separate `progress listener` for rescan, that includes error and completion callbacks along with the progress callback.